hollow/src/main/java/com/netflix/hollow/api/consumer/index/UniqueKeyIndex.java

```diff
@@ -157,13 +158,32 @@ public T apply(Q key) {
      * @return the unique object
      */
     public T findMatch(Q key) {
-        Object[] keyArray = matchFields.stream().map(mf -> mf.extract(key)).toArray();
 
-        int ordinal = hpki.getMatchingOrdinal(keyArray);
-        if (ordinal == -1) {
-            return null;
+        Object[] keyArray = new Object[matchFields.size()];
```

Comment:
Lets make this an instance variable

Reply:
done

Reply:
Thanks, we can also move the initialization up into the constructor so that the array is initialized once instead of per call to findMatch

Reply:
sorry. missed this. For some reason, I thought matchField size would vary for every findMatch call so I let it stay here.